stdlib/public/TensorFlow/OpaqueHandles.swift

```diff
@@ -16,61 +16,117 @@
 
 import CTensorFlow
 
-/// `ResourceHandle` is the type used by ops and the `#tfop()` syntax to
-/// represent TensorFlow "resource" values.  It exists only to represent edges
-/// in TensorFlow graphs, and has no host-side representation (and thus no
-/// methods).
-public final class ResourceHandle {
-  private init() {
-    fatalError("ResourceHandle is a marker type that can never be created")
-  }
-}
-
-/// `VariantHandle` is the type used by ops and the `#tfop()` syntax to
-/// represent TensorFlow "variant" values.
-/// TODO: rename this source file as this is no longer "opaque".
-public final class VariantHandle {
-  public let cTensorHandle: CTensorHandle
-
+/// A class that implements common functionality for resource and variant handles.
+public class ResourceVariantHandleBase {
+	public let cTensorHandle: CTensorHandle
+	public let cIsResource: Bool
+	
   @usableFromInline
-  init(owning cTensorHandle: CTensorHandle) {
+  init(owning cTensorHandle: CTensorHandle, cIsResource: Bool) {
     self.cTensorHandle = cTensorHandle
+		self.cIsResource = cIsResource
   }
 
   deinit {
     debugLog("De-initializing TensorHandle.")
     TFE_DeleteTensorHandle(cTensorHandle)
-    debugLog("Returning from deinit of VariantHandle.")
+    debugLog("Returning from deinit of handle.")
   }
-}
 
-extension VariantHandle : TensorSendableReceivable {
-  @inlinable
-  static func receiveFromAccelerator(
-    _ computation: _TensorComputation,
-    _ tensorID: Int
-  ) -> VariantHandle {
-    debugLog("Receiving variant tensor of id \(tensorID).")
+  static public func getTFDataTypeName(_ isResource: Bool) -> String {
+    return isResource ? "resource" : "variant"
+  }
+	
+	@inlinable
+	static func receiveTensorHandleHelper(
+		_ computation: _TensorComputation,
+    _ tensorID: Int,
+	  _ isResource: Bool) -> CTensorHandle {
```

Comment:
consider passing enum TF_RESOURCE / TF_VARIANT directly instead of the bool isResource. This might also facilitate future code refactoring / merging with sends/recvs code in TensorHandle (consider removing the `cIsResource` member property field).

Also, we might not need getTFDataTypeName() in that case. see https://github.com/apple/swift/blob/20e0cca669105d5590aeeb658690e0ddf2342234/stdlib/public/TensorFlow/CompilerRuntime.swift#L496. if really needed, we can write a util method that converts the enum to a human-readable string for the type name.